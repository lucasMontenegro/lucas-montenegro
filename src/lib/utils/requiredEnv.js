import globals from "lib/utils/globals"
export default function requiredEnv (key) {
  const { env } = globals.process
  if (key in env) {
    return env[key]
  }
  throw Error(`process.env.${key} is required`)
}