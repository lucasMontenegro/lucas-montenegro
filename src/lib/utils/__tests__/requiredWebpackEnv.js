import requiredEnv from "../requiredWebpackEnv"
describe(`../requiredEnv`, () => {
  jestUtils.describeDependencies({
    // usage: requiredWebpackEnv(process.env.FOO)
    // 'process.env.FOO' is replaced at compile-time
    deps: [`react-scripts`],
    relativeBasePath: __dirname,
  })
  describe(`requiredWebpackEnv (value is a string)`, () => {
    it(`should return the value`, () => {
      expect(requiredEnv(`foo`)).toBe(`foo`)
    })
  })
  describe(`requiredWebpackEnv (variable is undefined)`, () => {
    it(`should throw`, () => {
      expect(() => requiredEnv(undefined)).toThrow(`An environment variable is required`)
    })
  })
})