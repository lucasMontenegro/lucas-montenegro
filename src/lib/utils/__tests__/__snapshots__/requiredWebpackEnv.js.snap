// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`../requiredEnv external dependencies react-scripts should match the installed version 1`] = `"^3.3.1"`;
